qa/rpc-tests/wallet_golden_5_6_0.py

```diff
@@ -0,0 +1,225 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2020 The Zcash developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or https://www.opensource.org/licenses/mit-license.php .
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    NU5_BRANCH_ID,
+    assert_equal,
+    get_coinbase_address,
+    nuparams,
+    start_nodes,
+    stop_nodes,
+    wait_bitcoinds,
+    wait_and_assert_operationid_status,
+    persistent_cache_exists,
+    persist_node_caches,
+)
+from decimal import Decimal
+
+
+
+class WalletGoldenV5_6_0Test(BitcoinTestFramework):
+    def __init__(self):
+        super().__init__()
+        self.num_nodes = 4
+        # Start from the cache if it exists
+        if persistent_cache_exists('golden-v5.6.0'):
+            self.cache_behavior = 'golden-v5.6.0'
+
+    def setup_nodes(self):
+        return start_nodes(self.num_nodes, self.options.tmpdir, extra_args=[[
+            nuparams(NU5_BRANCH_ID, 201),
+        ]] * self.num_nodes)
+
+    def run_test(self):
+        # Establish the persistent cache state before running the main behavior
+        # of the test.
+        if not persistent_cache_exists('golden-v5.6.0'):
```

Comment:
```suggestion
        if self.cache_behavior != 'golden-v5.6.0':
```
(avoid checking whether the cache exists both in `__init__` and here)